tests/test_time_series/test_time_lag.py

```diff
@@ -0,0 +1,150 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from feature_engine.timeseries.forecasting import TimeSeriesLagTransformer
+_false_input_params = [
+    ([3, 2], "pizza", ["tango", "empanada"], 9),
+    ("mate", ["bombilla", "limpiado"], True, "cocinado"),
+    (True, False, "asado", [1984, 1999]),
+]
+
+
+@pytest.mark.parametrize(
+    "_variables, _periods, _freq, _drop_original",
+    _false_input_params
+)
+def test_raises_error_when_wrong_input_params(
+        _variables, _periods, _freq, _drop_original, df_time
+):
+    with pytest.raises(KeyError):
+        transformer1 = TimeSeriesLagTransformer(variables=_variables)
+        transformer1.fit(df_time)
+        transformer1.transform(df_time)
+    with pytest.raises(ValueError):
+        transformer2 = TimeSeriesLagTransformer(periods=_periods)
+        transformer2.fit(df_time)
+        transformer2.transform(df_time)
+    with pytest.raises((ValueError, TypeError)):
+        transformer3 = TimeSeriesLagTransformer(freq=_freq)
+        transformer3.fit(df_time)
+        transformer3.transform(df_time)
+    with pytest.raises(ValueError):
+        transformer4 = TimeSeriesLagTransformer(drop_original=_drop_original)
+        transformer4.fit(df_time)
+        transformer4.transform(df_time)
+
+
+def test_default_params():
```

Comment:
can we extend this test to evaluate multiple values for each parameter using parametrize? 

For variables try a string, and a list of strings and None. For periods test int, list of int and None, for freq, str, list of str and none.

Reply:
How do we test for multiple values when we are testing the default params? There's only one default value per param.

Reply:
I would say, let's change the test name and test the default paramaters and other permitted values for the params as well.